src/grammar_base.jl

```diff
@@ -230,6 +230,18 @@ function add_rule!(g::Grammar, e::Expr)
 	return g
 end
 
+"""
+Adds a probabilistic derivation rule.
+"""
+function add_rule!(g::Grammar, p::Real, e::Expr)
```

Comment:
Do we want this to be called `add_probabilistic_rule`? Just like with `parse_rule`/`parse_probabilistic_rule`?

Reply:
Here the multiple dispatch can handle the choice, whereas in the case of parsing rules it can't, since they have the exact same argument type.